addons/website_quote/static/src/js/website_quotation.js

```diff
@@ -17,26 +17,97 @@ if(!$('.o_website_quote').length) {
-                $('[data-id="total_amount"]>span').html(data[1]);
-            });
+            var order_id = href.match(/order_id=([0-9]+)/)[1];
+            var line_id = href.match(/update_line(_dict)?\/([0-9]+)/)[2];
+            var token = href.match(/token=([\w\d-]*)/)[1];
+
+            var callParams = {
+                'line_id': line_id,
+                'order_id': parseInt(order_id),
+                'token': token,
+                'remove': this.$el.is('[href*="remove"]'),
+                'unlink': this.$el.is('[href*="unlink"]')
+            };
+            this._callUpdateLineRoute(callParams).then(this._updateOrderValues.bind(this));
             return false;
         },
+        /**
+         * Only for compatibily reasons
+         * when server is not restarted
+         * and the route update_line_dict doesn't exist
+         */
+        _callUpdateLineRoute: function (params) {
+            var def = new $.Deferred();
+            ajax.jsonRpc("/quote/update_line_dict", 'call', params)
+                .then(def.resolve.bind(def))
+                .fail(function () {
+                    // Compatibility: the server may not have been restarted
+                    // So the real route may not exist
+                    ajax.jsonRpc("/quote/update_line", 'call', params)
+                        .then(def.resolve.bind(def))
+                        .fail(def.reject.bind(def));
+                });
+            return def;
+        },
+        _updateOrderValues: function (data) {
+            if(!data){
+                location.reload();
+            }
+
+            // Compatibility: the other route may have been called
+            var orderAmountTotal = Array.isArray(data) ? data[1] : data.order_amount_total;
+            var orderAmountUntaxed = data.order_amount_untaxed;
+            var orderAmountTax = data.order_amount_tax;
+
+            // Compatibility: the other route may have been called
+            var lineProductUomQty = Array.isArray(data) ? data[0] : data.order_line_product_uom_qty;
+            var linePriceTotal = data.order_line_price_total;
+            var linePriceSubTotal = data.order_line_price_subtotal;
+
+            this.$el.parents('.input-group:first').find('.js_quantity').val(lineProductUomQty);
+
+            var $parentTr = this.$el.parents('tr:first');
+            var $elPriceTotal = $parentTr.find('.oe_order_line_price_total .oe_currency_value');
+            var $elPriceSubTotal = $parentTr.find('.oe_order_line_price_subtotal .oe_currency_value');
+
+            // Compatibility: the module is not updgraded
+            // so classes in DOM are not present
+            if ( !$elPriceTotal.length && !$elPriceSubTotal.length ) {
+                $elPriceTotal = $elPriceSubTotal = $parentTr.find('.oe_currency_value').last();
+            }
+
+            if ($elPriceTotal.length && linePriceTotal) {
```

Comment:
If the price is 0 we still want to show it, so probably need to test `!== undefined` here. Same remark for all the other places were the price is tested.